timeboost-crypto/src/sg_encryption.rs

```diff
@@ -202,21 +202,19 @@ where
         );
         let pk_comb = comb_key.key.clone();
 
-        // Verify DLEQ proofs
-        let valid_shares: Vec<_> = dec_shares
-            .iter()
-            .filter_map(|share| {
+        // Verify DLEQ proofs to ensure correctness of the decryption share w_i
+        // keeping a list of faulty shares (their node indices) to blame
+        let (valid_shares, faulty_shares): (Vec<&Self::DecShare>, Vec<_>) =
+            dec_shares.iter().partition(|share| {
                 let (w, phi) = (share.w, share.phi.clone());
                 let u = pk_comb[share.index as usize];
```

Comment:
Not related to this PR, but maybe worth including: Using `Vec::get` instead of the index operator here is probably a good idea, given that these values are passed in as arguments.

Reply:
sounds reasonable, but this will spoil the `partition()` structure. I'll update the PR

Reply:
fixed in [68ee2df](https://github.com/EspressoSystems/timeboost/pull/342/commits/68ee2df52f3c78a10b6f8ca5ceccb6c0936434ff)